### PR TITLE
Reuse the dense log-depth tensor for SILog in `DepthLoss26`

### PR DESCRIPTION
**Pull request by JESUSROYETH**
`DepthLoss26` computes `log()` on the predicted/GT depth twice for the same valid pixels: once on a gathered subset for the SILog term, and again on the dense tensor a few lines later for the gradient-matching loss. The dense version is already needed for the pyramid, so SILog can just read from it instead of running its own `log()`.

Deleted: the separate `pred_valid`/`gt_valid` gather-then-log path for SILog (4 lines), reusing the dense `pred_log`/`gt_log` already computed for the gradient loss.

```python
# before
pred_valid = pred_depth[valid]
gt_valid = gt_depth[valid]
pred_valid = pred_valid.clamp(min=0.001)
log_diff = torch.log(pred_valid) - torch.log(gt_valid)
...
pred_log = torch.log(pred_depth.clamp(min=0.001))
gt_log = torch.log(gt_depth.clamp(min=0.001))

# after
pred_log = torch.log(pred_depth.clamp(min=0.001))
gt_log = torch.log(gt_depth.clamp(min=0.001))
log_diff = pred_log[valid] - gt_log[valid]
```

## Forward is exact, backward isn't

Checked both, not just forward, a fix touching a trainable path can change the graph even when the values agree. `torch.log(x)[mask]` and `torch.log(x[mask])` give the same values, `clamp(min=0.001)` before or after the gather is also the same thing for these pixels (values are already `> 0.001` there by construction of `valid`). Ran this on real tensors with `torch.equal`, forward loss is bit for bit identical.

Backward is not exact, and it shouldn't be. Before, `log()` ran as two independent nodes (one feeding SILog, one feeding the gradient loss), each with its own path back to `pred_depth.grad`. Now there is one `log()` node with two consumers, so autograd sums the two contributions in a different order. Floating-point addition isn't associative, so `pred.grad` differs by a few ULPs: max abs diff `7.45e-09`, relative `~1e-7`, float32 machine epsilon. Checked with `torch.testing.assert_close` (`rtol=1e-5, atol=1e-7`) instead of `torch.equal` for the gradient. This noise is already smaller than what batch sampling or AMP add to the gradient every step, so it shouldn't show up in training.

## Measured

Timed the full `DepthLoss26.__call__` (forward + backward, both loss terms, the 4-level gradient pyramid) on a realistic training batch, `bs=16, imgsz=640`, RTX 4060. This machine throttles under sustained load, so instead of a single before/after block I ran interleaved rounds (new, old, new, old, 4 rounds x 5 reps), reporting the minimum of each block since throttling only adds latency:

```
synthetic dense GT (valid=100%):  new min 50.26ms avg vs old min 51.58ms avg -> -2.6% avg per-round delta (4/4 rounds favour the new version)
synthetic sparse GT (valid=15%):  new min 20.09ms avg vs old min 20.20ms avg -> -0.5% avg per-round delta (4/4 rounds favour the new version)
```

Smaller than expected before measuring it .. the isolated SILog computation alone is ~10% faster, but it's a small slice of the full step once the gradient pyramid and the backward pass are included.

I didn't have a real depth GT sparsity number handy, so instead of guessing I also ran this against actual KITTI GT: 16 real samples from `data_depth_selection` (the official KITTI validation-subset archive, converted through the same PNG->`.npy` pipeline `depth-kitti.yaml`'s own `download:` block uses) through the real `DepthDataset` train pipeline (`imgsz=768`, the resolution the published weights use). Measured `valid_frac≈0.071`, sparser than my synthetic guess above. Same picture, new min 27.77ms avg vs old min 27.88ms avg, `-0.4%` avg per-round delta over two independent runs (7 of 8 rounds favour the new code, one round in the second run came out +0.3% .. at this magnitude the effect is close to the machine's noise floor, so a round going the other way isn't surprising). Also re-checked forward on this real batch with the actual `DepthLoss26` class, old and new loaded straight from the real file content, not a hand-copy .. still exact.

All three configurations were faster with the new code on average, real KITTI data included.

## Validation

`tests/test_python.py -k depth` (17 tests, all pass unchanged), including `test_v26_depth_loss_lower_lambda_penalizes_scale_error_more`, which already exercises `DepthLoss26` directly. No new test: existing coverage already runs this path, and the change has no new failure mode to guard against — it's a reordering of already-computed values, verified equivalent above.

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
Reuses dense log-depth tensors in `DepthLoss26` to compute both SILog and multi-scale gradient losses efficiently. ⚡

### 📊 Key Changes
- Computes clamped logarithms for prediction and ground-truth depth once across the full tensor.
- Reuses the dense `pred_log` and `gt_log` tensors for SILog masking and gradient matching.
- Removes duplicate logarithm calculations while preserving valid-pixel filtering for SILog.

### 🎯 Purpose & Impact
- Reduces redundant computation in depth-loss evaluation and simplifies the implementation.
- Keeps the gradient loss supplied with dense log-depth inputs as required for multi-scale matching.
- Maintains existing numerical safeguards, masking behavior, and loss outputs while potentially improving efficiency. 🚀

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @JESUSROYETH, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! To ensure a seamless integration of your work, please review the following checklist:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_ — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing). This automated message confirms the initial PR review steps; an Ultralytics engineer will assist with the review. Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

**Review by UltralyticsAssistant — APPROVED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Clean refactor. The dense log-depth tensors are required by the gradient loss already, and valid ground-truth pixels are strictly above the clamp threshold, so reusing them for SILog preserves the loss semantics while removing redundant work.

**Comment by glenn-jocher** [comment]
Thanks for the detailed validation. Reusing `pred_log` and `gt_log` is a clean, minimal optimization that preserves SILog semantics, and all CI checks are passing.

**Comment by glenn-jocher** [comment]
Thanks for the unusually thorough forward/backward validation and the net-negative implementation. The remaining blocker is user impact: the realistic KITTI result is about 0.4 percent, close to the reported machine noise floor, while the change intentionally alters floating-point gradient accumulation order in a training loss. Under the current strict gate, removing duplicate work alone is not enough when the end-to-end benefit is not distinguishable from noise. Please provide reproducible end-to-end training throughput or memory evidence showing a meaningful improvement on realistic data; otherwise I recommend closing this as technically sound but not materially beneficial to users.
